src/senselab/audio/tasks/features_extraction/praat_parselmouth.py

```diff
@@ -1,143 +1,1377 @@
-"""This module contains functions that extract features from audio files using the PRAAT library."""
+"""This module contains functions that extract features from audio files using the PRAAT library.
 
-from typing import Dict, List
+The initial implementation of this features extraction was started by Nicholas Cummins
+from King's College London and has since been further developed and maintained
+by the senselab community.
+"""
 
-import parselmouth
-from parselmouth.praat import call
+import inspect
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+
+import numpy as np
+import parselmouth  # type: ignore
+import pydra  # type: ignore
 
 from senselab.audio.data_structures import Audio
+from senselab.utils.data_structures import logger
 
 
-def get_audios_durations(audios: List[Audio]) -> List[Dict[str, float]]:
-    """Returns the duration of the Audio objects."""
+def get_sound(audio: Union[Path, Audio], sampling_rate: int = 16000) -> parselmouth.Sound:
+    """Get a sound object from a given audio file or Audio object.
 
-    def get_audio_duration(audio: Audio) -> Dict[str, float]:
-        def _get_duration(sound: parselmouth.Sound) -> float:
-            return call(sound, "Get total duration")
+    Args:
+        audio (Union[Path, Audio]): A path to an audio file or an Audio object.
+        sampling_rate (int, optional): The sampling rate of the audio. Defaults to 16000.
 
-        waveform = audio.waveform
-        sampling_rate = audio.sampling_rate
-        sound = parselmouth.Sound(waveform, sampling_frequency=sampling_rate)
-        duration = _get_duration(sound)
-        return {"duration": duration}
+    Returns:
+        parselmouth.Sound: A Parselmouth Sound object.
```

Comment:
should we try to harmonize sound/audio objects?

Reply:
this implementation is because parselmouth.Sound is not pickable, which is a problem with pydra

Reply:
we should test this directly with `cloudpickle`, which is what pydra uses. any known reasons for this? perhaps something we could address upstream depending on the reason.

Reply:
you might be able to write a hash function for parselmouth.Sound, similar to how we had to write one for Torch.Tensor so that Audios would work. I know that was a different issue but it possibly could be solved this way?

Reply:
i remember i spent way too much than I wanted trying to do it, but with no success. same thing with opensmile's smile object, which is why opensmile feats extraction is not parallelized. but I agree with you that it would be great it get it done. would be a great optimization